mara_pipelines/commands/files.py

```diff
@@ -57,21 +57,37 @@ def db_alias(self):
         return self._db_alias or config.default_db_alias()
 
     def shell_command(self):
-        copy_from_stdin_command = mara_db.shell.copy_from_stdin_command(
-            self.db_alias(), csv_format=self.csv_format, target_table=self.target_table,
-            skip_header=self.skip_header,
-            delimiter_char=self.delimiter_char, quote_char=self.quote_char,
-            null_value_string=self.null_value_string, timezone=self.timezone)
-        if not isinstance(mara_db.dbs.db(self.db_alias()), mara_db.dbs.BigQueryDB):
-            return \
-                f'{uncompressor(self.compression)} "{pathlib.Path(config.data_dir()) / self.file_name}" \\\n' \
-                + (f'  | {shlex.quote(sys.executable)} "{self.mapper_file_path()}" \\\n'
-                   if self.mapper_script_file_name else '') \
-                + ('  | sort -u \\\n' if self.make_unique else '') \
-                + '  | ' + copy_from_stdin_command
+        file_path = pathlib.Path(config.data_dir()) / self.file_name
+        db = mara_db.dbs.db(self.db_alias())
+        if isinstance(db, mara_db.dbs.SQLServerDB):
```

Comment:
That this generic command needs multiple branches for specific DBs is IMO a sign that the abstraction here is not yet right. If someone has a DB which is more similar to BQ/SQLServer, you are out of luck. Any chance to clean this up? E.g. move the specifics to a singledispatch functions (if BQ and SQL Server actually have some similarity) and either branch on some flag or simply call first one singledispatch function and if that raises NotImplmentedError try another?

Reply:
I agree. I did some researches. `functools` doesn't have a `singledispatch` for methods. I found serveral solutions [at stack overflow](https://stackoverflow.com/questions/24601722/how-can-i-use-functools-singledispatch-with-instance-methods).

What is your opinion, should we use an on implementation (first comment) or the module [`multipledispatch`](https://pypi.org/project/multipledispatch/)? @jankatins 


I think about adding a new method `read_file_command(self, db: object)` which is then called like this:
``` python
def shell_method(self):
    return self.read_file_command(mara_db.dbs.db(self.db_alias()))
```
`read_file_command` would then have a default implementation using `copy_from_stdin_command(...)` and have custom implementations for BigQueryDB and SQLServerDB

Reply:
I would just move the things to a function and pass whatever is needed as context to the function `whatever(source DB, ...)` and use `@singledispatch` or multidispatch if source + target are stuff to dispatch on (hopefully not). E.g. 

```
@singledispatch
def read_file_command(db_alias:str):
    return read_file_command(mara_db.dbs.db(db_alias))

@read_file_command.register(DB)
def read_file_command(db_alias:str):
    return ... current default

@read_file_command.register(BigQueryDB)
def read_file_command(db_alias:str):
    return ... BQ specific ...
````

?